File: spa-backend/src/main/java/Service/UserService.java
package Service;

import org.springframework.stereotype.Service;

import Model.User;
import Repo.UserRepo;

@Service
public class UserService {
	private UserRepo userRepo;
	
	public void addUser(String userName, String email, String password) {
		User newUser = new User(userName, email, password);
		userRepo.save(newUser);
	}

}

File: GroupAngularSPA/src/app/core/services/user.service.ts
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { User } from 'src/app/models/User';
const httpOptions = {
  headers: new HttpHeaders({
    'Content-Type': 'application/json',
  })
}
@Injectable({
  providedIn: 'root'
})
export class UserService {
  springUrl = 'http://localhost:8080';

  constructor(private http: HttpClient) { }

  registerUser(newUser : User):Observable<any>{
    //const user = {id: 0, username:"oo",email:"lsdjf",password:"sdkfjksd" };
    //const myJSON = JSON.stringify(user);
    //console.log(myJSON);
    return this.http.post<User>(`${this.springUrl}/newUser`, newUser , httpOptions);
  }
}
